lib/lz4frame.c

```diff
@@ -1554,7 +1556,7 @@ size_t LZ4F_decompress(LZ4F_dctx* dctx,
                 }   }
 
                 srcPtr += sizeToCopy;
-                dstPtr += sizeToCopy;
+                dstPtr = (BYTE*) ((uptrval) dstPtr + sizeToCopy);
```

Comment:
```
usan-log:../lib/lz4frame.c:1557:24: runtime error: applying zero offset to null pointer
```

Since we have `if (dstPtr == NULL)` in the previous block, this should be fixed as the following

```C
                if (dstPtr == NULL) {
                    sizeToCopy = 0;
                } else {
                    ...
                    if (dctx->frameInfo.blockMode == LZ4F_blockLinked) {
                       ...
-               }   }
+                   }
+                   dstPtr += sizeToCopy;
+               }

                srcPtr += sizeToCopy;
-               dstPtr += sizeToCopy; /* [usan] ../lib/lz4frame.c:1557:24: runtime error: applying zero offset to null pointer */
```